src/app/ReviewService/ReviewPrompter.Android.cs

```diff
@@ -1,26 +1,56 @@
 ﻿#if __ANDROID__
 using System;
-using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Plugin.StoreReview;
 using ReviewService.Abstractions;
 
 namespace ReviewService;
 
 /// <summary>
 /// Android implementation of <see cref="IReviewPrompter"/>.
 /// </summary>
-public sealed class ReviewPrompter : IReviewPrompter
+public sealed class ReviewPrompter : IReviewPrompter, IDisposable
 {
 	private readonly Android.OS.Handler _handler = new(Android.OS.Looper.MainLooper);
 
+	private readonly ILogger _logger;
+
+	public ReviewPrompter(ILogger<ReviewPrompter> logger)
+	{
+		_logger = logger ?? NullLogger<ReviewPrompter>.Instance;
+	}
+
 	/// <inheritdoc/>
 	public Task TryPrompt()
 	{
-		_handler.Post(() =>
+		try
 		{
-			// TODO.
-		});
-		throw new NotImplementedException();
+			_logger.LogDebug("Trying to prompt star rating to user.");
+
+			_handler.Post(async () =>
```

Comment:
Is this an async void?

Reply:
I don't think you have a choice, it have to be an action and it can't return a value. That's why I did not flag it.